app/src/controllers/reservation/services/ReservationService.ts

```diff
@@ -247,39 +247,92 @@ const ReservationService: ReservationServiceInterface = {
   },
 
   async updateReservation(user, shopId, reservationId, reservationDate, clientId, menuId, stylistId) {
-    if (user.role.slug === RoleSlug.SHOP_STAFF && !await isUserOwnedShop(user.id, shopId)) {
-      Logger.debug('Shop is not owned by user')
-      throw new AuthorizationError()
+    // check if shop exists
+    const shopDetails = await ShopRepository.fetchShopDetailsForReservation(shopId)
+    if (!shopDetails) {
+      Logger.debug('Shop does not exist')
+      throw new NotFoundError()
     }
 
-    const reservationExists = await ReservationRepository.reservationExists(reservationId)
-    if (!reservationExists) {
+    // check if reservation exists
+
+    const currentReservation = await ReservationRepository.fetchShopReservation(shopId, reservationId)
+    if (!currentReservation) {
       Logger.debug('Reservation does not exist')
       throw new NotFoundError()
     }
 
+    // check if new reservation is valid
+
+    if (reservationDate < new Date() || reservationDate === currentReservation.reservationDate) {
+      Logger.debug('Invalid date, earlier than today')
+      throw new NotFoundError()
+    }
```

Comment:
same date should be allowed because the client id or something else could be changed

Reply:
that is true i just have to check for time lol

Reply:
plus you should use today.ts lib for the new Date part. it's in UTC so it's kinda not correct